### ISSUE TITLE
Problem saving floats in multilang enviroment

### ISSUE THREAD

**Issue by lukaskleinschmidt** [getkirby/kirby#1422]
**Describe the bug**
Somehow as soon as I add a second language and try to save a `number` field or a `range` field for example float values get saved with a `,` instead of a `.`. German seems to be one of the languages that causes this problem.

**To Reproduce**
1. Fresh startkit with at least two languages (de, en)
2. Add a field like:
```yml
range:
  label: Range
  type: range
  step: 0.5
  min: 0
  max: 5
```
3. Save it in both languages

**Expected behavior**
No `,` when saving floating numbers.

**Kirby Version**
3.0.0 and 3.0.1-rc.1

**Console output**
The `PATCH` request from the panel is ok. So the panel is not the problem herei guess.

**Desktop (please complete the following information):**
 - Windows 10
 - Firefox 66.0b2 

**Additional context**
I tried it locally with a fresh startkit (3.0.0) and the project I am working on (3.0.1-rc.1) with the internal php server. And on a server for the staging (don't know the speccs right now).

**Comment by lukaskleinschmidt** [comment]
Perhaps related https://github.com/getkirby/kirby/issues/1264

**Comment by bastianallgeier** [comment]
✅ This is now fixed on the 3.0.2 branch.

**Comment by lukaskleinschmidt** [comment]
@bastianallgeier I am not really convinced of the solution to be honest. It kinda works because the input from the panel is always prepared by the `Form` class. But it only works for the pages `update()` method and then only if the saved key matches a field defined in the blueprint with the proper field type. Using the `save()` method would mean that I am responsible for formatting floats to a language agnostic string.

If I throw a array of data at a page I would expect proper float transformation (programmatically creating/updating pages for example) independently if there is a field definition within a blueprint or not.
My PR did not fix that either. But for my understanding this should be handled at best somewhere within the `Data` class directly or at least on `Page` level perhaps [here](https://github.com/getkirby/kirby/blob/master/src/Cms/ModelWithContent.php#L130).

**Comment by ralfgoeke** [comment]
I just installed the 3.0.2 pre release. Now fields of type number are stored as string in yaml. 

3.0.1: 
`mynumber: 42`

3.0.2:
`mynumber: "42"`

Guess it has something to do with this issue in a way

**Comment by bastianallgeier** [comment]
The biggest issue is that PHP is fucking stupid in here and floatval and (float) both are language-specific. That's a nightmare and I don't really know how to solve that. Converting float values to strings was my best guess, but I agree that it is not perfect. argh!!! Maybe we need to delay this further until we find a good way. IMHO we can't solve it in the Data handler because we never know there if the current value should be a float or not.

**Comment by lukasbestle** [comment]
Your solution to convert the floats to strings is definitely the right one. After all, that's what PHP does internally as well (but just in the wrong way as you wrote).

I agree with @lukaskleinschmidt that this should be solved as low-level as possible, meaning as close to the actual storage layer as possible. A string is only needed right before the data is stored to file, before that we shouldn't convert anything to string.

The Data handler *does* know if the value is a float (it can check with `is_float()`). Of course this will only work when *writing* data, but after all writing is the only issue here. Parsing that number again will work just fine at a higher level in the code and does *not* need to be handled by the Data handler.

Basically it's like this: The Data classes are responsible to convert any kind of data into a formatted string (be it YAML or KirbyData). YAML has a syntax that allows to differentiate strings and numbers (the quote `"`), KirbyData does not. But even with that difference, the task to fit numbers into the data should still be handled in the data handler, even with KirbyData. Otherwise we are always going to have inconsistencies between YAML (which handles that as well) and KirbyData.

So my opinion is that we should move the conversion directly to the [`Data\Txt` class, where we already have special handling for arrays](https://github.com/getkirby/kirby/blob/3.0.1/src/Data/Txt.php#L48).

**Comment by bastianallgeier** [comment]
Ok, I solved it in the Data handler and in the YAML encoder.
